The Big Guide to Web Components Built With React/The Big Guide to Web Components Built With React.md

```diff
@@ -0,0 +1,434 @@
+
+Instead, we decided to have our Components delivered with static file names (something like `shopping-component.js`), and we would rely on the [`ETag` Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) and the built-in browser behaviors around it to manage cache busting.
+
+To allow `ETag` headers to work, the server that delivers the static files simply needs to generate an `ETag` value for each new version of the file to compare against when the browser requests the static resource, and send that on the response with the content. When the browser sends a request for a file with an `ETag` Header value that matches the file on the server, the server returns a `304 Not Modified` response, and, the browser will fall back to its cached resource. When the `ETag` values are mismatched, the server will send the new version of the file along with a `200 OK` response.
+
+In our case, most of our applications are built with [.NET Core](https://docs.microsoft.com/en-us/dotnet/core/). In that framework, the [`useStaticFiles` middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-3.1#serve-files-inside-of-web-root) does the right thing with `ETag` headers out of the box. It will return a new `ETag` value for changed files, and will return a `304 Not Modified` response when the requested asset is unchanged.
+
+There are tradeoffs with utilizing the `Etag` header to manage caching:
+
+* The browser has to make a request to the server to confirm the `ETag` value.
+* There isn't a one-size solution to utilize `Cache-Control` headers.
+
+There are some mitigating factors there, though:
+
+* The size of the 304 response is a few hundred bytes, which is typically minuscule to your actual bundle size (generally some tens-to-hundreds of kilobytes).
+* If you're delivering a static file where you have a good understanding of the change rate (do we deploy new code every day, week, month, etc?), then you can set up `Cache-Control` headers to meet the needs of your specific case.
+
+Understanding how we can deploy these files with a static name and consistent cache busting is really good, but in itself, it's not enough to give us the independence for teams that we're looking for. For that, teams need to be able to deploy their applications independently, without the need to notify other teams of a change, or to step on the toes of other teams by deploying through the same deployment pipeline.
+
+#### Deploying Independently
+
+There are a number of ways to solve the problem of isolated, independent deployments, but our solution is to utilize right-hand routing (configured in a load balancer) to determine which servers a particular request should go to.
+
+In a traditional or monolithic application, traffic might move a little bit like this:
+
+![Diagram describing all traffic from base-domain.com being sent to a single primary server, no matter the path that is being requested.](./monolithic-routing.png)
```

Comment:
~The diagrams render as string literals not as diagrams.~  Diagrams are rendered in the final output, but not when viewing the file somehow.

Reply:
It looks like that's because it's a relative path, and those images don't exist relative to this pull request, but when you look at it in the source view, those images do exist relative to the file there.